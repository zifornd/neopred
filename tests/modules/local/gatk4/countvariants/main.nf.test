nextflow_process {

    name "Test Process GATK4_COUNTVARIANTS"
    script "modules/local/gatk4/countvariants/main.nf"
    process "GATK4_COUNTVARIANTS"
/*
// Missing output file(s) `*_counts` expected by process `GATK4_COUNTVARIANTS (SRR8281226)`
//Test will fail due to the file naming convention
    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of([
                    [ id:'test' ], 
                    [ file("$projectDir/tests/test_result/gatk/test.selected.vcf.gz", checkIfExists: true) ]
                ])

                input[1] = Channel.of([
                    [ id:'test' ], 
                    [ file("$projectDir/tests/test_result/gatk/test.selected.vcf.gz.tbi", checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assert process.success
            //assert process.out.counts
            assert process.out.versions
            assert snapshot(process.out).match()
        }

    }
*/
    test("Should run with options \"-stub\"") {
        options "-stub"
        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of([
                    [ id:'test' ], 
                    [ file("$projectDir/tests/test_result/gatk/test.selected.vcf.gz", checkIfExists: true) ]
                ])

                input[1] = Channel.of([
                    [ id:'test' ], 
                    [ file("$projectDir/tests/test_result/gatk/test.selected.vcf.gz.tbi", checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert process.out.counts
            assert process.out.versions
            assert snapshot(process.out).match()
        }

    }

}
