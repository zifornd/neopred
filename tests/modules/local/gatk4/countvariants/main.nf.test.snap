{
    "Should run with options \"-stub\"": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_counts:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c26f5d77c30929ba22c1f936590b62f0"
                ],
                "counts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_counts:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c26f5d77c30929ba22c1f936590b62f0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-08T06:14:52.29111"
    }
}